react/javascript/test/HighlightTest.tsx

```diff
@@ -74,74 +77,47 @@ describe('HighLight', () => {
-        .map((span) => span.innerHTML)
-        .join('')
-
-      assert.equal(
-        highlighted,
-        '<span>&lt;span&gt;Given&lt;/span&gt; a passed </span><mark>step</mark>'
-      )
-    })
-
-    it('also highlight the tags', () => {
-      const document = renderHighlight(
-        '<strong>Given</strong> a strong step',
-        'strong'
-      )
-      const highlighted = Array.from(
-        document.querySelectorAll('#content .highlight')
-      )
-        .map((span) => span.innerHTML)
-        .join('')
+  it('puts <mark> around matches in markdown', () => {
+    const document = renderHighlight(
+      '* This is\n* a bullet list',
+      'bullet',
+      true
+    )
+    const highlighted = Array.from(
+      document.querySelectorAll('#content mark')
+    ).map((span) => span.textContent)
 
-      assert.equal(
-        highlighted,
-        '<span>&lt;</span><mark>strong</mark><span>&gt;Given&lt;/</span><mark>strong</mark><span>&gt; a </span><mark>strong</mark><span> step</span>'
-      )
-    })
+    assert.deepStrictEqual(highlighted, ['bullet'])
   })
 
-  context('when htmlContent is set to true', () => {
-    it('keeps the HTML content', () => {
-      const document = renderHighlight(
-        '<em>Given</em> a passed step',
-        'step',
-        true
-      )
-      const highlighted = Array.from(
-        document.querySelectorAll('#content .highlight')
-      )
-        .map((span) => span.innerHTML)
-        .join('')
-
-      assert.equal(highlighted, '<em>Given</em> a passed <mark>step</mark>')
-    })
+  it('does not render <script> tags in markdown', () => {
+    const document = renderHighlight(
+      'Failed XSS: <script>alert("hello")</script>',
+      'alert hello',
+      true
+    )
+    const html = document.querySelector('#content').innerHTML
+    // Script tags will be removed (rather than escaped). Ideally we'd *escape* them to &lt;script&gt;.
+    assert.deepStrictEqual(html, '<div class="highlight"><p>Failed XSS: <mark>alert</mark>("<mark>hello</mark>")</p></div>')
+  })
 
-    it('does not highlight the tags', () => {
-      const document = renderHighlight(
-        '<strong>Given</strong> a strong step',
-        'strong',
-        true
-      )
-      const highlighted = Array.from(
-        document.querySelectorAll('#content .highlight')
-      )
-        .map((span) => span.innerHTML)
-        .join('')
+  it('renders <section> tags in markdown', () => {
+    const document = renderHighlight(
+      'We *like* other HTML tags:\n\n<section>hello</section>',
+      null,
+      true
+    )
+    const html = document.querySelector('#content').innerHTML
+    assert.deepStrictEqual(html, '<div class="highlight"><p>We <em>like</em> other HTML tags:</p><section>hello</section></div>')
+  })
 
-      assert.equal(
-        highlighted,
-        '<strong>Given</strong> a <mark>strong</mark> step'
-      )
-    })
+  it('does not render JavaScript event handlers on tags in markdown', () => {
```

Comment:
@vincent-psarga does this address your concern about javascript injection in element event handlers?

I assume it's more thoroughly tested in `react-markdown`...